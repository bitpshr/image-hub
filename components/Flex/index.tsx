import { CSSProperties, FunctionComponent, memo } from "react";

export interface FlexProps {
	alignContent?: CSSProperties["alignContent"];
	alignItems?: CSSProperties["alignItems"];
	alignSelf?: CSSProperties["alignSelf"];
	basis?: CSSProperties["flexBasis"];
	className?: string;
	direction?: CSSProperties["flexDirection"];
	display?: CSSProperties["display"];
	flex?: CSSProperties["flex"];
	flow?: CSSProperties["flexFlow"];
	grow?: CSSProperties["flexGrow"];
	justifyContent?: CSSProperties["justifyContent"];
	maxWidth?: CSSProperties["maxWidth"];
	minWidth?: CSSProperties["minWidth"];
	order?: CSSProperties["order"];
	shrink?: CSSProperties["flexShrink"];
	width?: CSSProperties["width"];
	wrap?: CSSProperties["flexWrap"];
}

const Flex: FunctionComponent<FlexProps> = memo((props) => (
	<div
		className={props.className}
		style={{
			...(props.alignContent ? { alignContent: props.alignContent } : {}),
			...(props.alignSelf ? { alignSelf: props.alignSelf } : {}),
			...(props.basis ? { flexBasis: props.basis } : {}),
			...(props.flex ? { flex: props.flex } : {}),
			...(props.flow ? { flexFlow: props.flow } : {}),
			...(props.grow ? { flexGrow: props.grow } : {}),
			...(props.justifyContent ? { justifyContent: props.justifyContent } : {}),
			...(props.maxWidth ? { maxWidth: props.maxWidth } : {}),
			...(props.minWidth ? { minWidth: props.minWidth } : {}),
			...(props.order ? { order: props.order } : {}),
			...(props.shrink ? { flexShrink: props.shrink } : {}),
			...(props.width ? { width: props.width } : {}),
			...(props.wrap ? { flexWrap: props.wrap } : {}),
			alignItems: props.alignItems || "flex-start",
			display: props.display || "flex",
			flexDirection: props.direction || "row",
		}}
	>
		{props.children}
	</div>
));

export default Flex;
