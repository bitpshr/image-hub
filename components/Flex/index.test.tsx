import React from "react";
import renderer from "react-test-renderer";

import Flex from "./";

test("default render", () => {
	const component = renderer.create(
		<Flex grow={1} shrink={0} alignItems="center" />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
