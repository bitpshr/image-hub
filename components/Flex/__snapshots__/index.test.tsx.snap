// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default render 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "flexGrow": 1,
    }
  }
/>
`;
